src/main/java/com/thomashan/coup/action/state/WaitingMainActionState.java

```diff
@@ -89,7 +95,7 @@ public ActionState performAction(MainAction action) {
         checkActionPlayerIsSameAsStatePlayer(action);
         checkActionAllowable(action);
 
-        List<Action> newActionHistory = CollectionUtil.add(actionHistory, action);
+        ImmutableList<Action> newActionHistory = actionHistory.plus(action);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Found 'DU'-anomaly for variable 'newActionHistory' (lines '98'-'120').](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4509131)